File: jmagent/src/com/jmagent/models/Result.java
package com.jmagent.models;

public class Result implements Item {
    private Integer resultId = -1;
    private Profile profile = new Profile();
    private String code = "";
    private Company company = new Company();
    private String lastUpdatedDate = "";
    private String postedDate = "";
    private boolean applied = false;
    private String title = "";
    private String description = "";
    private String required = "";
    private String optional = "";
    private String benefits = "";
    private String otherDetails = "";
    private Integer minYearsNeeded = 0;
    private String sourceEndpoint = "";
    private String baseEndpoint = "";
    private ResultState state = ResultState.NEW;

    public Result() {

    }

    public Result(String json) {

    }

    public String toJsonString() {
        String result = "{";
        result += String.format("\"id\":\"%d\"", this.resultId);
        result += String.format(",\"code\":\"%s\"", this.code);
        result += String.format(",\"minYears\":\"%d\"", this.minYearsNeeded);
        result += String.format(",\"state\":\"%d\"", this.state.ordinal());
        result += String.format(",\"applied\":\"%s\"", this.applied ? "1" : "0");
        result += String.format(",\"title\":\"%s\"", this.title);
        result += String.format(",\"description\":\"%s\"", this.description);
        result += String.format(",\"required\":\"%s\"", this.required);
        result += String.format(",\"optional\":\"%s\"", this.optional);
        result += String.format(",\"benefits\":\"%s\"", this.benefits);
        result += String.format(",\"otherDetails\":\"%s\"", this.otherDetails);
        result += String.format(",\"lastUpdatedDate\":\"%s\"", this.lastUpdatedDate);
        result += String.format(",\"sourceEndpoint\":\"%s\"", this.sourceEndpoint);
        result += String.format(",\"company\":%s", this.company.toJsonString());
        result += "}";
        return result;
    }

    public void setID(Integer id) { this.resultId = id; }
    public void setProfile(Profile profile) { this.profile = profile; }
    public void setCode(String code) { this.code = code; }
    public void setCompany(Company company) { this.company = company; }
    public void setLastUpdatedDate(String lastUpdated) { this.lastUpdatedDate = lastUpdated; }
    public void setPostedDate(String postedDate) { this.postedDate = postedDate; }
    public void setApplied(boolean applied) { this.applied = applied; }
    public void setTitle(String title) { this.title = title; }
    public void setDescription(String description) { this.description = description; }
    public void setRequired(String required) { this.required = required; }
    public void setOptional(String optional) { this.optional = optional; }
    public void setBenefits(String benefits) { this.benefits = benefits; }
    public void setOtherDetails(String otherDetails) { this.otherDetails = otherDetails; }
    public void setMinYearsNeeded(Integer minYears) { this.minYearsNeeded = minYears; }
    public void setSourceEndpoint(String endpoint) { this.sourceEndpoint = endpoint; }
    public void setBaseEndpoint(String endpoint) { this.baseEndpoint = endpoint; }
    public void setState(ResultState state) { this.state = state; }
    public Integer getID() { return this.resultId; }
    public Profile getProfile() { return this.profile; }
    public String getCode() { return this.code; }
    public Company getCompany() { return this.company; }
    public String getLastUpdatedDate() { return this.lastUpdatedDate; }
    public String getPostedDate() { return this.postedDate; }
    public boolean getApplied() { return this.applied; }
    public String getTitle() { return this.title; }
    public String getDescription() { return this.description; }
    public String getRequired() { return this.required; }
    public String getOptional() { return this.optional; }
    public String getBenefits() { return this.benefits; }
    public String getOtherDetails() { return this.otherDetails; }
    public Integer getMinYearsNeeded() { return this.minYearsNeeded; }
    public String getSourceEndpoint() { return this.sourceEndpoint; }
    public String getBaseEndpoint() { return this.baseEndpoint; }
    public ResultState getState() { return this.state; }
}

File: jmserver/src/classes/jmserver/models/Result.go
package models
import (
	"../../common"
	"fmt"
	"strconv"
)

type Result struct {
	id              int            `json:id`
    profile         *Profile       `json:profile`
    code            string         `json:code`
    company         *Company       `json:company`
    lastUpdatedDate string         `json:lastUpdatedDate`
    postedDate      string         `json:postedDate`
    applied         bool           `json:applied`
    title           string         `json:title`
    description     string         `json:description`
    required        string         `json:required`
    optional        string         `json:optional`
    benefits        string         `json:benefits`
    otherDetails    string         `json:otherDetails`
    minYearsNeeded  int            `json:minYearsNeeded`
    sourceEndpoint  string         `json:sourceEndpoing`
    state           int            `json:state`
}

func (x *Result) ReloadFromJson(json string) {
    var dict = common.StrToDictionary([]byte(json));
    x.id, _ = strconv.Atoi(dict["id"].(string));
    x.profile = &Profile{};
	x.profile.ReloadFromJson(common.DictionaryToJsonString(dict["name"].(map[string]interface{})));
    x.code = dict["code"].(string);
    x.company = &Company{};
	x.company.ReloadFromJson(common.DictionaryToJsonString(dict["city"].(map[string]interface{})));
    x.lastUpdatedDate = dict["lastUpdatedDate"].(string);
    x.postedDate = dict["postedDate"].(string);
    var val, _ = strconv.Atoi(dict["applied"].(string));
	x.applied = val > 0;
    x.title = dict["title"].(string);
    x.description = dict["description"].(string);
    x.required = dict["required"].(string);
    x.optional = dict["optional"].(string);
    x.benefits = dict["benefits"].(string);
    x.otherDetails = dict["otherDetails"].(string);
	x.minYearsNeeded, _ = strconv.Atoi(dict["minYearsNeeded"].(string));
	x.sourceEndpoint = dict["sourceEndpoing"].(string);
	x.state, _ = strconv.Atoi(dict["state"].(string));
}

func (x *Result) ToJsonString() string {
    var result = "{";
    result += fmt.Sprintf("\"id\":\"%d\"", x.id);
    result += fmt.Sprintf(",\"profile\":%s", x.profile.ToJsonString());
    result += fmt.Sprintf(",\"code\":\"%s\"", x.code);
    result += fmt.Sprintf(",\"company\":%s", x.company.ToJsonString());
    result += fmt.Sprintf(",\"lastUpdatedDate\":\"%s\"", x.lastUpdatedDate);
    result += fmt.Sprintf(",\"postedDate\":\"%s\"", x.postedDate);
    result += fmt.Sprintf(",\"applied\":\"%d\"", x.applied);
    result += fmt.Sprintf(",\"title\":\"%s\"", x.title);
    result += fmt.Sprintf(",\"description\":\"%s\"", x.description);
    result += fmt.Sprintf(",\"required\":\"%s\"", x.required);
    result += fmt.Sprintf(",\"optional\":\"%s\"", x.optional);
    result += fmt.Sprintf(",\"benefits\":\"%s\"", x.benefits);
    result += fmt.Sprintf(",\"otherDetails\":\"%s\"", x.otherDetails);
	result += fmt.Sprintf(",\"minYearsNeeded\":\"%d\"", x.minYearsNeeded);
	result += fmt.Sprintf(",\"sourceEndpoint\":\"%s\"", x.sourceEndpoint);
	result += fmt.Sprintf(",\"state\":\"%d\"", x.state);
    result += "}";
    return result;
}
func (x *Result) SetID (a int) { x.id = a; }
func (x *Result) SetProifle (a *Profile) { x.profile = a; }
func (x *Result) SetCode (a string) { x.code = a; }
func (x *Result) SetCompany (a *Company) {x.company = a; }
func (x *Result) SetLastUpdatedDate (a string) { x.lastUpdatedDate = a; }
func (x *Result) SetPostedDate (a string) { x.postedDate = a; }
func (x *Result) SetApplied (a bool) { x.applied = a; }
func (x *Result) SetTitle (a string) { x.title = a; }
func (x *Result) SetDescription (a string) { x.description = a; }
func (x *Result) SetRequired (a string) { x.required = a; }
func (x *Result) SetOptional (a string) { x.optional = a; }
func (x *Result) SetBenefits (a string) { x.benefits = a; }
func (x *Result) SetMinYearsNeeded (a int) { x.minYearsNeeded = a; }
func (x *Result) SetSourceEndpoint (a string) { x.sourceEndpoint = a; }
func (x *Result) SetState (a int) { x.state = a; }
func (x Result) ID() (int) { return x.id; }
func (x Result) Profile() (*Profile) { return x.profile; }
func (x Result) Code() (string) { return x.code; }
func (x Result) Company() (*Company) { return x.company; }
func (x Result) LastUpdatedDate() (string) { return x.lastUpdatedDate; }
func (x Result) PostedDate() (string) { return x.postedDate; }
func (x Result) Applied() (bool) { return x.applied; }
func (x Result) Title() (string) { return x.title; }
func (x Result) Description() (string) { return x.description; }
func (x Result) Required() (string) { return x.required; }
func (x Result) Optional() (string) { return x.optional; }
func (x Result) Benefits() (string) { return x.benefits; }
func (x Result) OtherDetails() (string) { return x.otherDetails; }
func (x Result) MinYearsNeeded() (int) { return x.minYearsNeeded; }
func (x Result) SourceEndpoint() (string) { return x.sourceEndpoint; }
func (x Result) State() (int) { return x.state; }

